core/src/main/java/org/primefaces/extensions/component/keynote/Keynote.java

```diff
@@ -0,0 +1,613 @@
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.faces.FacesException;
+import javax.faces.application.ResourceDependency;
+import javax.faces.component.ContextCallback;
+import javax.faces.component.UIComponent;
+import javax.faces.component.UINamingContainer;
+import javax.faces.component.behavior.ClientBehaviorHolder;
+import javax.faces.component.visit.VisitCallback;
+import javax.faces.component.visit.VisitContext;
+import javax.faces.context.FacesContext;
+import javax.faces.event.AjaxBehaviorEvent;
+import javax.faces.event.BehaviorEvent;
+import javax.faces.event.FacesEvent;
+import javax.faces.event.PhaseId;
+
+import org.primefaces.component.api.PrimeClientBehaviorHolder;
+import org.primefaces.component.api.Widget;
+import org.primefaces.extensions.component.base.AbstractDynamicData;
+import org.primefaces.extensions.event.KeynoteEvent;
+import org.primefaces.extensions.model.common.KeyData;
+import org.primefaces.extensions.model.keynote.KeynoteItem;
+import org.primefaces.util.Constants;
+import org.primefaces.util.MapBuilder;
+
+@ResourceDependency(library = "primefaces", name = "jquery/jquery.js")
+@ResourceDependency(library = "primefaces", name = "jquery/jquery-plugins.js")
+@ResourceDependency(library = "primefaces", name = "core.js")
+@ResourceDependency(library = "primefaces-extensions", name = "primefaces-extensions.js")
+@ResourceDependency(library = "primefaces-extensions", name = "keynote/keynote.js")
+@ResourceDependency(library = "primefaces-extensions", name = "keynote/keynote.css")
+public class Keynote extends AbstractDynamicData implements Widget, ClientBehaviorHolder, PrimeClientBehaviorHolder {
+
+    public static final String COMPONENT_TYPE = "org.primefaces.extensions.component.Keynote";
+    public static final String COMPONENT_FAMILY = "org.primefaces.extensions.component";
+    public static final String DEFAULT_RENDERER = "org.primefaces.extensions.component.KeynoteRenderer";
+
+    private static final String DEFAULT_EVENT = "slideChanged";
+
+    private static final Map<String, Class<? extends BehaviorEvent>> BEHAVIOR_EVENT_MAPPING = MapBuilder.<String, Class<? extends BehaviorEvent>> builder()
+                .put("slideTransitionEnd", null)
+                .put(DEFAULT_EVENT, null)
+                .build();
+
+    private static final Collection<String> EVENT_NAMES = BEHAVIOR_EVENT_MAPPING.keySet();
+
+    private Map<String, UIKeynoteItem> items;
+
+    protected enum PropertyKeys {
+        //@formatter:off
+        widgetVar,
+        width,
+        height,
+        margin,
+        minScale,
+        maxScale,
+        autoSlide,
+        center,
+        controls,
+        disableLayout,
+        embedded,
+        loop,
+        navigationMode,
+        progress,
+        showNotes,
+        slideNumber,
+        touch,
+        transition,
+        transitionSpeed,
+        backgroundTransition,
+        theme,
+        library,
```

Comment:
@melloware what do you think of this way of offering themes to be loaded?

IMO we should just offer / support the bundled themes. And if you want something custom, simply create custom styles in a custom style sheet.